src/app/features/components/login/login.component.ts

```diff
@@ -32,21 +32,19 @@ export class LoginComponent implements OnInit {
     this.loginModel = this.loginForm.value;
     this.authenticationService.login(this.loginModel).subscribe((response: TokenModel) => {
       if(response!=null) {
-        this.redirectToViewBusinessTripsBasedOnRole(response);
+        localStorage.setItem('token', JSON.stringify({ token: response.token}));
```

Comment:
please remove this, you should use that in interceptor

Reply:
It proves that I need to set this here in order to decode the role of the user later; the interceptor will only read it after each request and add it to the header.

Reply:
You can use a service to receive the role directly.